drivers/serial/uart_nrfx_uart.c

```diff
@@ -1024,7 +1040,10 @@ static void uart_nrfx_set_power_state(struct device *dev, u32_t new_state)
 	if (new_state == DEVICE_PM_ACTIVE_STATE) {
 		uart_nrfx_pins_enable(dev, true);
 		nrf_uart_enable(uart0_addr);
-		nrf_uart_task_trigger(uart0_addr, NRF_UART_TASK_STARTRX);
+		if (RX_PIN_USED()) {
+			nrf_uart_task_trigger(uart0_addr,
+					      NRF_UART_TASK_STARTRX);
```

Comment:
UARTE driver needs similar change. I guess it would be okay to do this in this PR.